okhttp/src/main/kotlin/okhttp3/internal/socket/OkioSslSocketFactory.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2024 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package okhttp3.internal.socket
+
+import javax.net.ssl.SSLSocket
+import javax.net.ssl.SSLSocketFactory
+
+interface OkioSslSocketFactory {
+  fun createSocket(socket: OkioSocket): OkioSslSocket
+
+  fun createSocket(
+    socket: OkioSocket,
+    host: String,
```

Comment:
Should we combine these to InetSocketAddress?

Reply:
Also looking at UnixDomainSocketFactory, I wonder whether the type of Address should be more flexible. For UnixDomainSocketFactory, it's really a File, or Path, and we have a bunch of lying methods.

Not sure this is possible...